addons/mail/static/src/core/web_portal/message_model_patch.js

```diff
@@ -0,0 +1,11 @@
+import { Message } from "@mail/core/common/message_model";
+import { patch } from "@web/core/utils/patch";
+
+patch(Message.prototype, {
+    get canCopyLink() {
+        return super.canCopyLink && this.store.self.active !== false;
```

Comment:
This won't work for portal users regardless, as they will be active but can't access the backend.

But the real problem is that we direct to back-end rather than the appropriate page depending on user access, which was already known even before the PR was merged https://github.com/odoo/odoo/pull/152648#issuecomment-2265178899 and nothing was done about it.